### PR TITLE
GamePlay Replaces Board

### PR DESCRIPTION
**Pull request by kelseyroy**
Adds:
- GamePlay class
- New GamePlay functionality, including get next player, select spot, validate input
- GamePlay tests
- A board as data to TicTacToe

Removes:
- Board class, Board methods now exist in GamePlay
- Board test suite, Board tests now exist in GamePlay test suite

### CODE REVIEW

File: lib/game_play.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        valid_input?(spot)
+    end
+
+    private
+
+    def valid_input?(spot)
+        (1..9).to_a.include?(spot)
+    end
+
+end
```
Comment (Aoxian) [inline lib/game_play.rb:47]: 🔫🔫🔫

Comment (Aoxian) [inline lib/game_play.rb:47]: 🔫🔫🔫

File: lib/game_play.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+class GamePlay
```
Comment (t-keazirian) [inline lib/game_play.rb:1]: Nice approach - this makes sense!

File: lib/game_play.rb (hunk 1 above)
Comment (Aoxian) [inline lib/game_play.rb]: 🔫 🔫 🔫

**Review by Aoxian — APPROVED** [review]

File: lib/game_play.rb (hunk 1 above)
Comment (Aoxian) [reply lib/game_play.rb]: "No newline at end of file"

**Review by t-keazirian — APPROVED** [review]

File: lib/game_play.rb (hunk 1 above)
Comment (kelseyroy) [reply lib/game_play.rb]: Added a new line! Do you think I should squash valid_input? and validate_input into one method to make it drier? I might be moving that into a validate class pretty soon anyways.

Comment (Aoxian) [reply lib/game_play.rb]: yeah I would make it just one function.

**Comment by kelseyroy** [comment]
Am I approved to merge?

File: lib/game_play.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        return spot
+    end
+
+    def validate_input(spot)
+        is_valid = (1..9).to_a.include?(spot) 
+        raise ArgumentError, "Not a number between 1-9" unless is_valid 
+        
+        is_valid
+    end
```
Comment (Aoxian) [inline lib/game_play.rb]: Nitpick; always better to be explicit about your return values.

```suggestion
    def validate_input(spot)
        is_valid = (1..9).to_a.include?(spot) 
        raise ArgumentError, "Not a number between 1-9" unless is_valid 
        
        true
    end
```

Comment (kelseyroy) [reply lib/game_play.rb]: I like this! Much easier to understand.

**Pull request merged**